src/server/game/Scripting/ScriptMgr.h

```diff
@@ -834,22 +835,46 @@ class TC_GAME_API GroupScript : public ScriptObject
 
 class TC_GAME_API SceneScript : public ScriptObject
 {
+protected:
+
+    SceneScript(const char* name);
+
+public:
+    // Called when a player start a scene
+    virtual void OnSceneStart(Player* /*player*/, uint32 /*sceneInstanceID*/, SceneTemplate const* /*sceneTemplate*/) { }
+
+    // Called when a player receive trigger from scene
+    virtual void OnSceneTriggerEvent(Player* /*player*/, uint32 /*sceneInstanceID*/, SceneTemplate const* /*sceneTemplate*/, std::string const& /*triggerName*/) { }
+
+    // Called when a scene is canceled
+    virtual void OnSceneCancel(Player* /*player*/, uint32 /*sceneInstanceID*/, SceneTemplate const* /*sceneTemplate*/) { }
+
+    // Called when a scene is completed
+    virtual void OnSceneComplete(Player* /*player*/, uint32 /*sceneInstanceID*/, SceneTemplate const* /*sceneTemplate*/) { }
+
+};
+
+class TC_GAME_API AreaTriggerEntityScript : public ScriptObject
```

Comment:
Bad idea imho, there is no reason to add an "OnUpdate" function in AreaTriggerScript by example. This is two totally different things, even if they have close names